pkg/gadgets/seccomp/gadget.go

```diff
@@ -303,6 +303,14 @@ func generateSeccompPolicy(client client.Client, trace *gadgetv1alpha1.Trace, b
 	r := syscallArrToSeccompPolicy(profileName, b)
 	seccompProfileAddLabelsAndAnnotations(r, trace, fullPodName, containername)
 
+	if trace.Spec.OutputMode == "ExternalResource" {
+		// If output mode is ExternalResource, thus Output field will not be used,
+		// so we can use it to store the profile name where the seccomp profile will
+		// be stored.
```

Comment:
This is not always true. Users can specify the name by using the CLI flag `--profile-prefix` and I don't think we should modify the trace's fields specified by the user.

In addition, this approach does not support multiple profiles generation thus when we will support it (near future), we will have to change this approach. I suggest implementing it to already support multiple profiles.

One possible solution could be to retrieve the SeccompProfiles from the CLI using the `global-trace-id`? It is already present in the annotations of the SeccompProfiles we generate using the CLI.

What do you think?

Reply:
You are right and hopefully you caught it because I totally forgot about the `--profile-prefix` :sweat_smile:.
I will go with filtering seccomp profile based on their labels, thank you for this suggestion.